HAZIK/Hobot_Gyenes/gyenes_hobot_hazi.py

```diff
@@ -0,0 +1,350 @@
+import networkx as nx
+import random
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import re
+
+header = ["user_id", "gender", "region", "AGE"]
+
+
+def minta(df_node, df_attr):
+    
+    #kell neki adni headert
+
+    header = ["user_id", "gender", "region", "AGE"]
+    df_attr.columns=header
+    df_node.columns=["node_1", "node_2"]
+    
+    #egy pár régiót választunk, ez a minta lényege
+    
+    df_attr=df_attr[(df_attr.region=="bratislavsky kraj, bratislava - nove mesto") | (df_attr.region=='bratislavsky kraj, bratislava - karlova ves') | (df_attr.region=='bratislavsky kraj, bratislava - ruzinov')]
+    
+    #itt lesz pár merge
+    na=df_attr.merge(df_node, left_on="user_id", right_on="node_1", how="inner")
+    nana=pd.DataFrame()
+    nana["user_id"]=na["user_id"]
+    nana["gender"]=na["gender"]
+    nana=nana.merge(na, how="left", right_on="node_2", left_on="user_id")
+    nana = nana[nana['node_1'].notna()]
+    nana.drop_duplicates()
+    nana=nana[["node_1", "node_2", "gender_y", "gender_x", "AGE"]]
+    nana.columns=["node_1", "node_2", "gender_1", "gender_2", "age_1"]
+    nana=nana.merge(df_attr, how="left", right_on="user_id", left_on="node_2")
+    nana=nana[["node_1", "node_2", "gender_1", "gender_2", "age_1", "AGE"]]
+    nana.columns=["node_1", "node_2", "gender_1", "gender_2", "age_1", "age_2"]
+    nana=nana.drop_duplicates()
+
+
+
+    
+    return nana
+
+
+
+
+
+def generate_attribute_data(data, G):
+    da1=pd.DataFrame(G.nodes.data("gender_1"), columns=['Node',"Gender"])
+    da2=pd.DataFrame(G.nodes.data("age_1"), columns=['Node',"Age"])
+    da3=pd.DataFrame(G.degree(), columns=['Node',"Degree"])
+    da4=pd.Series(nx.average_neighbor_degree(G)).to_frame().reset_index()
+    da4.columns = ['Node',"Avg_Neighbor_Degree"]
+    da5=pd.Series(nx.clustering(G)).to_frame().reset_index()
+    da5.columns = ['Node',"Clustering_Coeff"]
+    da6=pd.Series(nx.triangles(G)).to_frame().reset_index()
+    da6.columns = ['Node',"Num_Triangles"]
+    data=pd.merge(data, da1, how="inner", on="Node")
+    data=pd.merge(data, da2, how="inner", on="Node")
+    data=pd.merge(data, da3, how="inner", on="Node")
+    data=pd.merge(data, da4, how="inner", on="Node")
+    data=pd.merge(data, da5, how="inner", on="Node")
+    data=pd.merge(data, da6, how="inner", on="Node")
+    
+    return data
+
+def make_community(G):
+    #a függvény lényege, hogy feltérképezzük a legközelebb álló "klasztereket" Louvain metódussal
+    import community
+    bestcomm_dict = community.best_partition(G)
+    maxQ_L = community.modularity(bestcomm_dict,G)
+    print("Klaszterek száma:", max(bestcomm_dict.values())+1, "Louvain modularitás: %.3f"%maxQ_L)
+    
+    return bestcomm_dict
+
+def add_community(G, bestcomm_dict, data):
+    # a függvény célja, hogy hozzáadjuk a gráf attributumokat tároló dataframehez a nodeokhoz tartozó communitiket
+    commune=[]
+    nodes=[]
+    for node in G.nodes():
```

Comment:
itt legyártod ugyanazt a dictionaryt ami már megvan egy for ciklus segítségével. brrrrrrrrr